NPE when interrupting the extension creation process

If I do:
```
mvn io.quarkus.platform:quarkus-maven-plugin:2.7.4.Final:create-extension
```
Then ctrl + c, ctrl +c when the questions are asked, I end up with:
```
Caused by: java.lang.NullPointerException
    at io.quarkus.devtools.commands.CreateExtension.resolveExtensionId (CreateExtension.java:286)
    at io.quarkus.devtools.commands.CreateExtension.prepare (CreateExtension.java:202)
    at io.quarkus.devtools.commands.CreateExtension.execute (CreateExtension.java:281)
    at io.quarkus.maven.CreateExtensionMojo.execute (CreateExtensionMojo.java:241)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
```

We should probably handle that more gracefully and throw an error and exit.

--- a/devtools/maven/src/main/java/io/quarkus/maven/components/Prompter.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/components/Prompter.java
@@ -9,6 +9,7 @@ import java.util.function.Consumer;
 import org.aesh.readline.Readline;
 import org.aesh.readline.ReadlineBuilder;
 import org.aesh.readline.tty.terminal.TerminalConnection;
+import org.aesh.terminal.tty.Signal;
 
 /**
  * Prompt implementation.
@@ -49,6 +50,7 @@ public class Prompter {
             return;
         }
         final TerminalConnection connection = new TerminalConnection();
+        connection.setSignalHandler(interruptionSignalHandler());
         try {
             read(connection, ReadlineBuilder.builder().enableHistory(false).build(), prompts.iterator());
             connection.openBlocking();
@@ -68,5 +70,16 @@ public class Prompter {
                 read(connection, readline, prompts);
             }
         });
+    }
+
+    private Consumer<Signal> interruptionSignalHandler() {
+        return new Consumer<Signal>() {
+            @Override
+            public void accept(Signal signal) {
+                if (signal == Signal.INT) {
+                    throw new RuntimeException("Process interrupted");
+                }
+            }
+        };
     }
 }

